### PR TITLE
Feature/fix annotation grouping

### PR DESCRIPTION
**Pull request by singularo**
This should solve the UI deployment/layout issues on production 3.5. To test, you will need to change dsh.project to have:
```minishift start --openshift-version "v1.5.1"```
again, doh.

* Add some support for PATCH to the client
* Make the content type different for PATCH
* Add a parameter to the service calls so that its possible to tell if this is the 'main' service, or an additional one.
* Add annotations to the parent service via the PATCH request
* Add app name to all services created/updated.

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -497,14 +506,53 @@ public function createService(string $name, string $deployment_name, int $port,
       ],
     ];
 
-    return $this->request($resourceMethod['action'], $uri, $service);
+    $annotations = ['app' => $app_name];
+    $this->applyAnnotations($service, $annotations);
+
+    $result = $this->request($resourceMethod['action'], $uri, $service);
+    if ($result && $app_name != $name) {
+      $this->groupService($app_name, $name);
```
Comment (pingers) [inline src/Client.php]: Add a @todo/warning about the potential race condition.

File: src/Client.php [hunk 2]
```diff
@@ -1288,4 +1336,5 @@ private function apiCall(string $method, $name = '', $label = NULL) {
     return $this->request($resourceMethod['action'], $uri, [], $query);
   }
 
+
```
Comment (pingers) [inline src/Client.php]: BALEET IT!

File: src/Client.php [hunk 3]
```diff
@@ -497,14 +506,53 @@ public function createService(string $name, string $deployment_name, int $port,
...
 
   /**
    * {@inheritdoc}
    */
-  public function updateService(string $name, array $data) {
-    // TODO: Implement updateService() method.
+  public function groupService(string $app_name, string $name) {
+    $resourceMethod = $this->getResourceMethod(__METHOD__);
+    $uri = $this->createRequestUri($resourceMethod['uri'], [
+      'name' => (string) $app_name,
```
Comment (pingers) [inline src/Client.php]: Superfluous cast. The method signature enforces string via type hinting.

**Comment by pingers** [comment]
Couple fixes and :ship:

File: src/Client.php (hunk 3 above)
Comment (singularo) [reply src/Client.php]: This is a copy pasta, the patten is repeated heaps, fixing the one i did, leaving the rest for later.

**Pull request merged**